app/lib/donations/controllers/DonorsChooseDonationController.js

```diff
@@ -58,6 +65,59 @@ function DonorsChooseDonationController() {
 DonorsChooseDonationController.prototype.resourceName = 'donors-choose';
 
 /**
+ * Optional method to start the donation flow. Validates that this user is
+ * allowed to make a donation.
+ *
+ * @param request
+ *   Express Request object
+ * @param response
+ *   Express Response object
+ */
+DonorsChooseDonationController.prototype.start = function(request, response) {
+  var phone;
+  var configId;
+
+  // Find the user
+  phone = smsHelper.getNormalizedPhone(request.body.phone);
+  configId = request.query.id;
+  userModel.findOne({phone: phone}, onUserFound);
+
+  response.send();
+
+  // Callback after user is found
+  function onUserFound(err, doc) {
+    var config;
+    var donationsCount;
+    var i;
+
+    if (err) {
+      logger.error(err);
+    }
+
+    donationsCount = 0;
+    if (doc && doc.donations) {
+      for (i = 0; i < doc.donations.length; i++) {
+        if (configId == doc.donations[i].config_id) {
+          donationsCount = doc.donations[i].count;
+          break;
+        }
+      }
+    }
+
+    config = app.getConfig(app.ConfigName.DONORSCHOOSE, configId);
+
+    // If user has not hit the max limit, start the donation flow.
+    if (!config.max_donations_allowed || donationsCount < config.max_donations_allowed) {
+      sendSMS(phone, config.max_donations_reached_oip);
+    }
+    // Otherwise, send an error message
+    else {
+      sendSMS(phone, config.start_donation_flow);
```

Comment:
Hypothetical case: 

If a user hasn't played the game before and thus has no user doc, but wants to make a donation. 

If that user texts in the `DONATE` keyword, they' would get trapped in this flow--since no user doc is found, the `start_donation_flow` error message would be sent. 

How should we handle? Should we just create a user doc for them? (To allow people who haven't played the game to donate?)